### PR TITLE
PT-6479 Added ReviewCount

### PR DESCRIPTION
**Pull request by OlegoO**
## Description
Added ReviewCount to RatingProductDto and RatingStoreDto.
Added ProdutId to RatingStoreDto.
Fixed Response model for productRatingInStore.
Fixed Sonar warnings.

### Jira-link:

https://virtocommerce.atlassian.net/browse/PT-6479

### CODE REVIEW

**Review by Vectorfield4 — APPROVED** [review]

File: tests/VirtoCommerce.CustomerReviews.Test/CustomerReviewsTests.cs [hunk 1]
```diff
@@ -53,7 +52,7 @@ public async Task CanDoCRUDandSearch()
 
             // Read non-existing item
 
-            result = await CustomerReviewService().GetByIdsAsync(new[] { CustomerReviewId });
+            result = await CustomerReviewService().GetAsync(new List<string>() { CustomerReviewId });
```
Comment (pushnitsa) [inline tests/VirtoCommerce.CustomerReviews.Test/CustomerReviewsTests.cs:55]: Use ICrudService<>.GetByIdAsync() instead

File: tests/VirtoCommerce.CustomerReviews.Test/VirtoCommerce.CustomerReviews.Test.csproj [hunk 2]
```diff
@@ -4,7 +4,7 @@
     <noWarn>1591</noWarn>
   </PropertyGroup>
   <ItemGroup>
-    <PackageReference Include="coverlet.collector" Version="3.1.0">
+    <PackageReference Include="coverlet.collector" Version="3.1.1">
```
Comment (pushnitsa) [inline tests/VirtoCommerce.CustomerReviews.Test/VirtoCommerce.CustomerReviews.Test.csproj:7]: Why did you up this dependency?

File: src/VirtoCommerce.CustomerReviews.Data/Services/RatingService.cs [hunk 3]
```diff
@@ -62,11 +64,19 @@ private async Task Calculate(string storeId, string[] productIds)
...
-            List<RatingEntity> entities = new List<RatingEntity>();
+            var entities = new List<RatingEntity>();
             foreach (var productStore in reviews.GroupBy(r => new { r.ProductId, r.StoreId }))
             {
-                decimal rating = calculator.Calculate(productStore.Select(r => r.Rating).ToArray());
-                entities.Add(new RatingEntity() { ProductId = productStore.Key.ProductId, StoreId = productStore.Key.StoreId, Value = rating });
+                var storeReviews = productStore.Select(r => r.Rating).ToArray();
+                var storeTotalRating = calculator.Calculate(storeReviews);
+
+                entities.Add(new RatingEntity()
```
Comment (pushnitsa) [inline src/VirtoCommerce.CustomerReviews.Data/Services/RatingService.cs:73]: Remove "()"

File: src/VirtoCommerce.CustomerReviews.Data/Services/RatingService.cs [hunk 4]
```diff
@@ -19,13 +20,14 @@ public class RatingService : IRatingService
     {
         private readonly Func<ICustomerReviewRepository> _repositoryFactory;
         private readonly IEnumerable<IRatingCalculator> _ratingCalculators;
-        private readonly IStoreService _storeService;
-        private readonly IStoreSearchService _storeSearchService;
+        private readonly ICrudService<Store> _storeService;
+        private readonly ISearchService<StoreSearchCriteria, StoreSearchResult, Store> _storeSearchService;
```
Comment (pushnitsa) [inline src/VirtoCommerce.CustomerReviews.Data/Services/RatingService.cs:26]: Remove blank line

File: src/VirtoCommerce.CustomerReviews.Data/Migrations/20220131080404_AddReviewCount.cs [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+#nullable disable
+
+namespace VirtoCommerce.CustomerReviews.Data.Migrations
+{
+    public partial class AddReviewCount : Migration
+    {
+        protected override void Up(MigrationBuilder migrationBuilder)
+        {
+            migrationBuilder.AddColumn<int>(
```
Comment (pushnitsa) [inline src/VirtoCommerce.CustomerReviews.Data/Migrations/20220131080404_AddReviewCount.cs:11]: Why do we need to store Review count? It is a countable value, isn't it?

File: tests/VirtoCommerce.CustomerReviews.Test/CustomerReviewsTests.cs (hunk 1 above)
Comment (OlegoO) [reply tests/VirtoCommerce.CustomerReviews.Test/CustomerReviewsTests.cs:55]: Original tests work with the collection of reviews in results. I just fix the obsolete warning.

File: tests/VirtoCommerce.CustomerReviews.Test/VirtoCommerce.CustomerReviews.Test.csproj (hunk 2 above)
Comment (OlegoO) [reply tests/VirtoCommerce.CustomerReviews.Test/VirtoCommerce.CustomerReviews.Test.csproj:7]: Good question. I don't know. Maybe VS 2022 did it automatically. Should I revert it back

**Review by Vectorfield4 — APPROVED** [review]

**Pull request merged**